pkg/yang/yangtype.go

```diff
@@ -266,11 +270,14 @@ func (y *YangType) Equal(t *YangType) bool {
 		!ssEqual(y.POSIXPattern, t.POSIXPattern),
 		len(y.Range) != len(t.Range),
 		!y.Range.Equal(t.Range),
-		!tsEqual(y.Type, t.Type):
+		!tsEqual(y.Type, t.Type),
+		!cmp.Equal(y.Enum, t.Enum, cmp.Comparer(func(t, u EnumType) bool {
```

Comment:
This is interesting -- prior to this change, I presume that we were de-duplicating enumerations because we weren't comparing the actual `EnumType` that was within the `YangType`, is that right?

Reply:
Yes it doesn't check either the name nor the definition of the enum, so the comparison might return true and it gets discarded.